File: csharp/dirty/fr.soat.cleancode/BlockChain.cs
using System.Collections.Generic;

namespace fr.soat.cleancode
{
   public class BlockChain
    {
       public Dictionary<string, Block> blocks = new Dictionary<string, Block>();


       public BlockChain()
        {
            Block initialBlock = new Block(null, "Le 09 Avril, cet exercice a été donné à Paris au meetup SOAT");
            blocks.Add(initialBlock.hash, initialBlock);
        }

        public bool AddBlock(Block newBlock)
        {
            if (!CheckHashValidity(newBlock))
            {
                return false;
            }
            blocks.Add(newBlock.hash, newBlock);
            CleanBlockChain();
            return true;
        }

        public bool CheckHashValidity(Block newBlock)
        {
            if (newBlock.hash.Equals(newBlock.parentHash))
                return false;
                else
                {
                    //TODO 1
                }
            return true;
        }

        public Block GetLastBlock()
        {
            //TODO 2
            return null;
        }

        public void CleanBlockChain()
        {
            //TODO 3
        }

    }
}

File: java/dirty/src/main/java/fr/cleancode/BlockChain.java
package fr.cleancode;

import java.util.HashMap;

class BlockChain {

    HashMap<String,Block> blocks = new HashMap<>();

    BlockChain() {
        Block initialBlock = new Block(null, "Le 09 Avril, cet exercice a été donné à Paris au meetup SOAT");
        blocks.put(initialBlock.hash, initialBlock);
    }


    boolean addBlock(Block newBlock) {
        if (!checkHashValidity(newBlock)) {
            return false;
        }
        blocks.put(newBlock.hash, newBlock);
        cleanBlockChain();
        return true;
    }

    boolean checkHashValidity(Block newBlock) {

        if (newBlock.hash.equals(newBlock.parentHash))
            return false;
            else {
                //TODO 1
            }
        return true;
    }

    Block getLastBlock() {
        //TODO 2
        return null;
    }

    void cleanBlockChain() {
        //TODO 3
    }

}
